### PR TITLE
Service and Task controllers

### PR DESCRIPTION
**Pull request by diemonster**
**What does this pull request do?**
Adds implementation at the controller layer for Services and Task entities. 

Adds initial interfaces for Service and Task providers

**How should this be tested?**
Navigate to ./api and execute go test ./...

**Checklist**
- [x] Unit tests

### CODE REVIEW

File: api/controllers/service.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return fireball.NewJSONResponse(202, model)
+}
+
+func (s *ServiceController) DeleteService(c *fireball.Context) (fireball.Response, error) {
+	id := c.PathVariables["id"]
+	if err := s.ServiceProvider.Delete(id); err != nil {
```
Comment (sesh-kebab) [inline api/controllers/service.go]: Service deletion should be via a job right?

```
type JobType int64

const (
	DeleteEnvironmentJob JobType = 1 + iota
	DeleteServiceJob
	DeleteLoadBalancerJob
	DeleteTaskJob
	CreateTaskJob
)
```

File: api/controllers/service_test.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+}
+
+func TestDeleteService(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockService := mock_provider.NewMockServiceProvider(ctrl)
+	controller := NewServiceController(mockService)
+
+	mockService.EXPECT().
```
Comment (sesh-kebab) [inline api/controllers/service_test.go]: Same as above - should be done via a job request I would've thought.

File: api/controllers/task.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return fireball.NewJSONResponse(202, model)
+}
+
+func (t *TaskController) DeleteTask(c *fireball.Context) (fireball.Response, error) {
+	id := c.PathVariables["id"]
+	if err := t.TaskProvider.Delete(id); err != nil {
```
Comment (sesh-kebab) [inline api/controllers/task.go]: Deletion of a Task should also happen via a job request?

Comment (zpatrick) [inline api/controllers/task.go]: I think we're still a unclear on how tasks will look. This may look radically different later on

Comment (zpatrick) [inline api/controllers/task.go]: But yeah - more likely than not it'll be in a job

Comment (diemonster) [inline api/controllers/task.go]: Lets revisit this when we figure out how `task` will work in this refactor?

**Pull request merged**